opentelemetry-exporter-geneva/geneva-uploader/src/common.rs

```diff
@@ -0,0 +1,235 @@
+//! Common utilities and validation functions shared across the Geneva uploader crate.
+
+use reqwest::header::{HeaderMap, HeaderValue, ACCEPT, USER_AGENT};
+use thiserror::Error;
+
+/// Common validation errors
+#[derive(Debug, Error)]
+pub(crate) enum ValidationError {
+    #[error("Invalid user agent prefix: {0}")]
+    InvalidUserAgentPrefix(String),
+}
+
+pub(crate) type Result<T> = std::result::Result<T, ValidationError>;
+
+// Validates a user agent prefix for HTTP header compliance
+// Validation Rules:
+// - Must contain only ASCII printable characters (0x20-0x7E)
+// - Must not contain control characters (especially \r, \n, \0)
+// - Must not exceed 200 characters in length
+// - Must not be empty or only whitespace
+pub(crate) fn validate_user_agent_prefix(prefix: &str) -> Result<()> {
+    if prefix.trim().is_empty() {
+        return Err(ValidationError::InvalidUserAgentPrefix(
+            "User agent prefix cannot be empty or only whitespace".to_string(),
+        ));
+    }
+
+    if prefix.len() > 200 {
+        return Err(ValidationError::InvalidUserAgentPrefix(format!(
+            "User agent prefix too long: {len} characters (max 200)",
+            len = prefix.len()
+        )));
+    }
+
+    // Check for invalid characters
+    for (i, ch) in prefix.char_indices() {
+        match ch {
+            // Control characters that would break HTTP headers
+            '\r' | '\n' | '\0' => {
+                return Err(ValidationError::InvalidUserAgentPrefix(format!(
+                    "Invalid control character at position {i}: {ch:?}"
+                )));
+            }
+            // Non-ASCII or non-printable characters
+            ch if !ch.is_ascii() || (ch as u8) < 0x20 || (ch as u8) > 0x7E => {
+                return Err(ValidationError::InvalidUserAgentPrefix(format!(
+                    "Invalid character at position {i}: {ch:?} (must be ASCII printable)"
+                )));
+            }
+            _ => {} // Valid character
+        }
+    }
+
+    Ok(())
+}
+
+// Builds a standardized User-Agent header for Geneva services
+// TODO: Update the user agent format based on whether custom config will come first or later
+// Current format:
+// - If prefix is None or empty: "GenevaUploader/0.1"
+// - If prefix is provided: "{prefix} (GenevaUploader/0.1)"
+pub(crate) fn build_user_agent_header(user_agent_prefix: Option<&str>) -> Result<HeaderValue> {
+    let prefix = user_agent_prefix.unwrap_or("");
+
+    // Validate the prefix if provided
+    if !prefix.is_empty() {
+        validate_user_agent_prefix(prefix)?;
+    }
+
+    let user_agent = if prefix.is_empty() {
+        "GenevaUploader/0.1".to_string()
+    } else {
+        format!("{prefix} (GenevaUploader/0.1)")
+    };
+
+    HeaderValue::from_str(&user_agent).map_err(|e| {
```

Comment:
It looks like this method does the ASCII characters validation check anyway so we could get avoid checking that ourselves.